x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/utils/persistence/MlBulkIndexer.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.ml.utils.persistence;
+
+import org.elasticsearch.action.bulk.BulkRequest;
+import org.elasticsearch.action.index.IndexRequest;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.index.IndexingPressure;
+
+import java.util.Objects;
+import java.util.function.Consumer;
+
+/**
+ * A helper class that gathers index requests in bulk requests
+ * that do exceed a 1000 operations or half the available memory
+ * limit for indexing.
+ */
+public class MlBulkIndexer implements AutoCloseable {
```

Comment:
There is no ML-specific code in this class. Would it make sense to rename it to `BulkIndexer`?

Reply:
I wanted to avoid calling it plainly `BulkIndexer` as it sounds too much like a basic ES component. I decided to prefix with `Ml` to indicate this is a utility designed to serve purposes within the ML plugin. If we end up using it in other places then we shall move it and rename it accordingly.